diff -uarN naoqi_driver/CMakeLists.txt naoqi_driver_dvlab/CMakeLists.txt
--- naoqi_driver/CMakeLists.txt	2021-02-13 14:21:12.200464764 +0100
+++ naoqi_driver_dvlab/CMakeLists.txt	2021-02-13 14:02:51.812022858 +0100
@@ -41,6 +41,7 @@
   src/subscribers/teleop.cpp
   src/subscribers/moveto.cpp
   src/subscribers/speech.cpp
+  src/subscribers/turnhead.cpp
   )
 
 set(
diff -uarN naoqi_driver/src/naoqi_driver.cpp naoqi_driver_dvlab/src/naoqi_driver.cpp
--- naoqi_driver/src/naoqi_driver.cpp	2021-02-13 14:21:12.212465046 +0100
+++ naoqi_driver_dvlab/src/naoqi_driver.cpp	2021-02-13 14:16:10.373420699 +0100
@@ -63,6 +63,7 @@
 #include "subscribers/teleop.hpp"
 #include "subscribers/moveto.hpp"
 #include "subscribers/speech.hpp"
+#include "subscribers/turnhead.hpp"
 
 
 /*
@@ -931,6 +932,7 @@
   registerSubscriber( boost::make_shared<naoqi::subscriber::TeleopSubscriber>("teleop", "/cmd_vel", "/joint_angles", sessionPtr_) );
   registerSubscriber( boost::make_shared<naoqi::subscriber::MovetoSubscriber>("moveto", "/move_base_simple/goal", sessionPtr_, tf2_buffer_) );
   registerSubscriber( boost::make_shared<naoqi::subscriber::SpeechSubscriber>("speech", "/speech", sessionPtr_) );
+  registerSubscriber( boost::make_shared<naoqi::subscriber::TurnheadSubscriber>("turnhead", "/naoqi_driver/turnhead", sessionPtr_) );
 }
 
 void Driver::registerService( service::Service srv )
